src/main/scala/diplomacy/LazyModule.scala

```diff
@@ -112,28 +168,43 @@ abstract class LazyModule()(implicit val p: Parameters)
     children.filter(!_.omitGraphML).foreach { c => c.edgesGraphML(buf, pad) }
   }
 
+  /** execute function to [[children]].
+    * @param iterfunc function to run
+    * */
   def nodeIterator(iterfunc: (LazyModule) => Unit): Unit = {
     iterfunc(this)
     children.foreach( _.nodeIterator(iterfunc) )
   }
 
+  /** get [[children]] */
   def getChildren = children
 }
 
 object LazyModule
 {
+  /** current [[LazyModule]] scope, specifically, it is a stack of [[LazyModule]].
+    * it will be dynamically set by [[LazyScope.apply]] and [[LazyModule.apply]],
+    * */
   protected[diplomacy] var scope: Option[LazyModule] = None
+  /** index of [[LazyModule]], notice there is no No.0 module. */
   private var index = 0
 
+  /** add a [[LazyModule]]
+    * @param bc [[LazyModule]] needed to be instantiated.
+    * @param valName [[ValName]] used to naming this val.
+    * @param sourceInfo [[SourceInfo]] to generate this [[LazyModule]]
+    * */
   def apply[T <: LazyModule](bc: T)(implicit valName: ValName, sourceInfo: SourceInfo): T = {
-    // Make sure the user put LazyModule around modules in the correct order
-    // If this require fails, probably some grandchild was missing a LazyModule
-    // ... or you applied LazyModule twice
+    /** Make sure the user puts LazyModule around modules in the correct order
+      * If this require fails, probably some grandchild was missing a LazyModule
+      * or you applied LazyModule twice.
+      * */
     require (scope.isDefined, s"LazyModule() applied to ${bc.name} twice ${sourceLine(sourceInfo)}")
     require (scope.get eq bc, s"LazyModule() applied to ${bc.name} before ${scope.get.name} ${sourceLine(sourceInfo)}")
+    /** [[LazyModule.scope]] stack pop. */
     scope = bc.parent
     bc.info = sourceInfo
-    if (!bc.suggestedNameVar.isDefined) bc.suggestName(valName.name)
+    if (bc.suggestedNameVar.isEmpty) bc.suggestName(valName.name)
```

Comment:
again i see this is the same functionally, but for the purposes of this PR, I suggest leaving the non-documentation changes out (ok I know i said to change some require messages too ;) )

Reply:
OK! I'll revoke these changes.